src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,116 @@
+package ui;
+
+import java.util.Scanner;
+
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import task.Task;
+import task.TaskList;
+
+/**
+ * Represents the user interface of the chatbot.
+ */
+public class Ui {
+    private final Scanner scanner;
+
+    public Ui() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Returns the welcome message.
+     */
+    public String showWelcome() {
+        return "Hello! I'm Buddy\nWhat can I do for you?";
+    }
+
+    /**
+     * Returns the goodbye message.
+     */
+    public String showGoodbye() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Returns an error message when loading tasks from file fails.
+     */
+    public String showLoadingError() {
+        return "Failed to load tasks from file!";
+    }
+
+    /**
+     * Returns an error message when saving tasks to file fails.
+     */
+    public String showSavingError() {
+        return "Failed to save tasks to file!";
+    }
+
+    /**
+     * Returns a message when a task is added.
+     */
+    public String showTaskAdded(Task task, int size) {
+        return "Got it. I've added this task:\n  " + task + "\nNow you have " + size + " tasks in the list.";
+    }
+
+    /**
+     * Returns a message when a task is marked as done.
+     */
+    public String showTaskMarked(Task task) {
+        return "Nice! I've marked this task as done:\n  " + task;
+    }
+
+    /**
+     * Returns a message when a task is unmarked.
+     */
+    public String showTaskUnmarked(Task task) {
```

Comment:
Maybe handling the response messages could be added to the individual command classes instead? The command classes could provide the response text and then call a function in the UI to display the message instead.